### PR TITLE
REST:  add the round number to algod box endpoint repsonse

### PR DESCRIPTION
**Pull request by bbroder-algo**
## Summary

Include the round that the value of the box's contents is pulled from to the response from v2/box endpoint

## Test Plan

Modified e2e to check rounds.

### CODE REVIEW

**Review by jannotti — DISMISSED** [review]
Seems like a good idea and correct implementation to me.

**Comment by bbroder-algo** [comment]
I made up the round numbers in the e2e test, they need to be replaced with the right answer.

**Comment by jannotti** [comment]
> I made up the round numbers in the e2e test, they need to be replaced with the right answer.

Will that even be possible?  Are these e2e tests run deterministically? (the e2e_subs are not)

**Comment by bbroder-algo** [comment]
Ran in sandbox, caught up to testnet, found an app with a box, ran box endpoint: 
```
bob@algobob sandbox #curl -i -H "X-Algo-API-Token:${AUTH_TOKEN}" "localhost:31200/v2/applications/200641557/box?name=b64:ABrYmUED76S3MLW1Lkb0mvlRD0YJD2rX2y5QIu%2FqXz09"
HTTP/1.1 200 OK
Content-Type: application/json; charset=UTF-8
Vary: Origin
Date: Thu, 27 Apr 2023 22:36:28 GMT
Content-Length: 120

{"name":"ABrYmUED76S3MLW1Lkb0mvlRD0YJD2rX2y5QIu/qXz09","round":29457190,"value":"AAAAAAAhQ6AAAAAAZD+e9AAAAAAABfXhAA=="}
bob@algobob sandbox #
```

**Review by ahangsu — APPROVED** [review]
seems right

File: daemon/algod/api/algod.oas3.yml [hunk 1]
```diff
@@ -1375,6 +1375,10 @@
             "pattern": "^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$",
             "type": "string"
           },
+          "round": {
+            "description": "The round for which this information is relevant",
+            "type": "integer"
+          },
```
Comment (bbroder-algo) [inline daemon/algod/api/algod.oas3.yml:1381]: @jasonpaulos would it be better to add the round in BoxResponse instead?  BoxResponse uses a schema reference though, and I seem to remember something about not being able to add to a schema reference.

Comment (jasonpaulos) [reply daemon/algod/api/algod.oas3.yml:1381]: No, this makes sense here. The schema ref that BoxResponse uses means it's really just an alias to Box. See our generated type here: https://github.com/algorand/go-algorand/blob/ebcd07b302d0f26d85ae538fbf1c2d620e7bed2e/daemon/algod/api/server/v2/generated/model/types.go#L858

File: test/e2e-go/restAPI/restClient_test.go [hunk 2]
```diff
@@ -1522,15 +1522,23 @@ end:
...
+		a.NoError(err)
 		boxResponse, err := testClient.GetApplicationBoxByName(uint64(createdAppID), boxTest.encodedName)
 		a.NoError(err)
+		currentRoundAfterBoxes, err := testClient.CurrentRound()
+		a.NoError(err)
 		a.Equal(boxTest.name, boxResponse.Name)
 		a.Equal(boxTest.value, boxResponse.Value)
+		// To reduce flakiness, only check the round from boxes is within a range.
+		a.GreaterOrEqual(boxResponse.Round, currentRoundBeforeBoxes)
+		a.LessOrEqual(boxResponse.Round, currentRoundAfterBoxes)
```
Comment (jasonpaulos) [inline test/e2e-go/restAPI/restClient_test.go:1541]: Nicely done

**Review by jasonpaulos — APPROVED** [review]

**Pull request merged**